// fs-extra-promise-es6/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.writeJson = exports.remove = exports.readJson = exports.outputJson = exports.outputFile = exports.move = exports.mkdirs = exports.ensureSymlink = exports.ensureLink = exports.ensureDir = exports.ensureFile = exports.emptyDir = exports.copy = exports.writeJsonSync = exports.walk = exports.removeSync = exports.readJsonSync = exports.outputJsonSync = exports.outputFileSync = exports.mkdirsSync = exports.ensureSymlinkSync = exports.ensureLinkSync = exports.ensureDirSync = exports.ensureFileSync = exports.emptyDirSync = exports.createOutputStream = exports.copySync = undefined;

var _nodePromiseEs = require('node-promise-es6');

var _fsExtra = require('fs-extra');

var fse = _interopRequireWildcard(_fsExtra);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

// export * from 'fs-extra';
/* eslint-disable */
var copySync = exports.copySync = fse.copySync;
var createOutputStream = exports.createOutputStream = fse.createOutputStream;
var emptyDirSync = exports.emptyDirSync = fse.emptyDirSync;
var ensureFileSync = exports.ensureFileSync = fse.ensureFileSync;
var ensureDirSync = exports.ensureDirSync = fse.ensureDirSync;
var ensureLinkSync = exports.ensureLinkSync = fse.ensureLinkSync;
var ensureSymlinkSync = exports.ensureSymlinkSync = fse.ensureSymlinkSync;
var mkdirsSync = exports.mkdirsSync = fse.mkdirsSync;
var outputFileSync = exports.outputFileSync = fse.outputFileSync;
var outputJsonSync = exports.outputJsonSync = fse.outputJsonSync;
var readJsonSync = exports.readJsonSync = fse.readJsonSync;
var removeSync = exports.removeSync = fse.removeSync;
var walk = exports.walk = fse.walk;
var writeJsonSync = exports.writeJsonSync = fse.writeJsonSync;
/* eslint-enable */

var copy = exports.copy = (0, _nodePromiseEs.promisify)(fse.copy);
var emptyDir = exports.emptyDir = (0, _nodePromiseEs.promisify)(fse.emptyDir);
var ensureFile = exports.ensureFile = (0, _nodePromiseEs.promisify)(fse.ensureFile);
var ensureDir = exports.ensureDir = (0, _nodePromiseEs.promisify)(fse.ensureDir);
var ensureLink = exports.ensureLink = (0, _nodePromiseEs.promisify)(fse.ensureLink);
var ensureSymlink = exports.ensureSymlink = (0, _nodePromiseEs.promisify)(fse.ensureSymlink);
var mkdirs = exports.mkdirs = (0, _nodePromiseEs.promisify)(fse.mkdirs);
var move = exports.move = (0, _nodePromiseEs.promisify)(fse.move);
var outputFile = exports.outputFile = (0, _nodePromiseEs.promisify)(fse.outputFile);
var outputJson = exports.outputJson = (0, _nodePromiseEs.promisify)(fse.outputJson);
var readJson = exports.readJson = (0, _nodePromiseEs.promisify)(fse.readJson);
var remove = exports.remove = (0, _nodePromiseEs.promisify)(fse.remove);
var writeJson = exports.writeJson = (0, _nodePromiseEs.promisify)(fse.writeJson);
