// dist-es6/lib/package-json.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _typeof2 = require('babel-runtime/helpers/typeof');

var _typeof3 = _interopRequireDefault(_typeof2);

var _deleteProperty = require('babel-runtime/core-js/reflect/delete-property');

var _deleteProperty2 = _interopRequireDefault(_deleteProperty);

var _getIterator2 = require('babel-runtime/core-js/get-iterator');

var _getIterator3 = _interopRequireDefault(_getIterator2);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function mapObj(mapper, obj) {
  return (0, _keys2.default)(obj).reduce(function (memo, key) {
    return (0, _assign2.default)(memo, (0, _defineProperty3.default)({}, key, mapper(obj[key], key)));
  }, {});
}

function removeUndefinedValues(obj) {
  var _iteratorNormalCompletion = true;
  var _didIteratorError = false;
  var _iteratorError = undefined;

  try {
    for (var _iterator = (0, _getIterator3.default)((0, _keys2.default)(obj)), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
      var key = _step.value;

      var value = obj[key];
      if (value === null || typeof value === 'undefined') {
        (0, _deleteProperty2.default)(obj, key);
      } else if ((typeof value === 'undefined' ? 'undefined' : (0, _typeof3.default)(value)) === 'object' && !Array.isArray(obj)) {
        removeUndefinedValues(value);
      }
    }
  } catch (err) {
    _didIteratorError = true;
    _iteratorError = err;
  } finally {
    try {
      if (!_iteratorNormalCompletion && _iterator.return) {
        _iterator.return();
      }
    } finally {
      if (_didIteratorError) {
        throw _iteratorError;
      }
    }
  }
}

var PackageJson = function () {
  function PackageJson(packageJson) {
    (0, _classCallCheck3.default)(this, PackageJson);

    removeUndefinedValues(packageJson);
    this.packageJson = packageJson;
  }

  (0, _createClass3.default)(PackageJson, [{
    key: 'toJSON',
    value: function toJSON() {
      return this.packageJson;
    }
  }, {
    key: 'map',
    value: function map(mapper, key) {
      var _this = this;

      if (key) {
        var value = this.packageJson[key];
        if (Array.isArray(value)) {
          return value.map(mapper);
        } else if (value !== null && (typeof value === 'undefined' ? 'undefined' : (0, _typeof3.default)(value)) === 'object') {
          return mapObj(mapper, value);
        } else if (value === null || typeof value === 'undefined') {
          return null;
        }

        return mapper(value);
      }

      var overrides = mapObj(function (subMapper, subKey) {
        if (typeof subMapper === 'function') {
          return _this.map(subMapper, subKey);
        }

        return subMapper;
      }, mapper);
      return new PackageJson((0, _assign2.default)({}, this.packageJson, overrides));
    }
  }, {
    key: 'moveTo',
    value: function moveTo(destinationPath) {
      function moveFile(filePath) {
        return _path2.default.relative(destinationPath, filePath);
      }
      return this.map({
        files: moveFile,
        main: moveFile,
        bin: moveFile
      });
    }
  }, {
    key: 'toProduction',
    value: function toProduction() {
      return this.map({
        files: null,
        scripts: function scripts(value, key) {
          if (key === 'prepublish') {
            return null;
          }
          return value;
        }
      });
    }
  }, {
    key: 'addBabelRuntime',
    value: function addBabelRuntime() {
      /* eslint-disable lines-around-comment, global-require */
      var _require$version$matc = require('babel/package.json').version.match(/^(\d+\.\d+)\.\d+$/);

      var _require$version$matc2 = (0, _slicedToArray3.default)(_require$version$matc, 2);

      var runtimeVersion = _require$version$matc2[1];
      /* eslint-enable lines-around-comment, global-require */

      return this.map({
        dependencies: (0, _assign2.default)({}, this.packageJson.dependencies, {
          'babel-runtime': '^' + runtimeVersion
        })
      });
    }
  }]);
  return PackageJson;
}();

exports.default = PackageJson;
