// dist-es6/lib/project.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

var _stringify = require('babel-runtime/core-js/json/stringify');

var _stringify2 = _interopRequireDefault(_stringify);

var _taggedTemplateLiteral2 = require('babel-runtime/helpers/taggedTemplateLiteral');

var _taggedTemplateLiteral3 = _interopRequireDefault(_taggedTemplateLiteral2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _entries = require('babel-runtime/core-js/object/entries');

var _entries2 = _interopRequireDefault(_entries);

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _regenerator = require('babel-runtime/regenerator');

var _regenerator2 = _interopRequireDefault(_regenerator);

var _asyncToGenerator2 = require('babel-runtime/helpers/asyncToGenerator');

var _asyncToGenerator3 = _interopRequireDefault(_asyncToGenerator2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _templateObject = (0, _taggedTemplateLiteral3.default)(['\n            #!/usr/bin/env node\n            \'use strict\';\n            require(\'register-module\')({\n              name: ', ',\n              path: ', ',\n              main: ', '\n            });\n            ', '(\'', '\');\n          '], ['\n            #!/usr/bin/env node\n            \'use strict\';\n            require(\'register-module\')({\n              name: ', ',\n              path: ', ',\n              main: ', '\n            });\n            ', '(\'', '\');\n          ']);

var _dedent = require('dedent');

var _dedent2 = _interopRequireDefault(_dedent);

var _directory = require('./directory');

var _directory2 = _interopRequireDefault(_directory);

var _packageJson = require('../lib/package-json');

var _packageJson2 = _interopRequireDefault(_packageJson);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Project = function () {
  function Project() {
    (0, _classCallCheck3.default)(this, Project);

    for (var _len = arguments.length, projectPath = Array(_len), _key = 0; _key < _len; _key++) {
      projectPath[_key] = arguments[_key];
    }

    this.directory = new (Function.prototype.bind.apply(_directory2.default, [null].concat(projectPath)))();
  }

  (0, _createClass3.default)(Project, [{
    key: 'packageJson',
    value: function () {
      var _ref = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee() {
        return _regenerator2.default.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                if (this.cachedPackageJson) {
                  _context.next = 4;
                  break;
                }

                _context.next = 3;
                return this.directory.readFile('package.json');

              case 3:
                this.cachedPackageJson = _context.sent;

              case 4:
                return _context.abrupt('return', this.cachedPackageJson);

              case 5:
              case 'end':
                return _context.stop();
            }
          }
        }, _callee, this);
      }));

      function packageJson() {
        return _ref.apply(this, arguments);
      }

      return packageJson;
    }()
  }, {
    key: 'linkBins',
    value: function () {
      var _ref2 = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee3() {
        var _this = this;

        var packageJson, nodeModules, bin;
        return _regenerator2.default.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.next = 2;
                return this.packageJson();

              case 2:
                packageJson = _context3.sent;
                _context3.next = 5;
                return this.directory.mkdir('node_modules');

              case 5:
                nodeModules = _context3.sent;
                _context3.next = 8;
                return nodeModules.mkdir('.bin');

              case 8:
                bin = _context3.sent;
                _context3.next = 11;
                return _promise2.default.all((0, _entries2.default)(Object(packageJson.bin)).map(function () {
                  var _ref3 = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee2(_ref4) {
                    var _ref5 = (0, _slicedToArray3.default)(_ref4, 2);

                    var binName = _ref5[0];
                    var binPath = _ref5[1];
                    var binContents;
                    return _regenerator2.default.wrap(function _callee2$(_context2) {
                      while (1) {
                        switch (_context2.prev = _context2.next) {
                          case 0:
                            _context2.next = 2;
                            return _this.directory.readFile(binPath);

                          case 2:
                            binContents = _context2.sent;
                            _context2.next = 5;
                            return bin.writeFile(binName, (0, _dedent2.default)(_templateObject, (0, _stringify2.default)(packageJson.name), (0, _stringify2.default)(_this.directory.join('src')), (0, _stringify2.default)(packageJson.main ? packageJson.main.replace('src/', '') : 'index.js'), binContents.indexOf('#!/usr/bin/env node') >= 0 ? 'require' : "require('dist-es6/lib/run').module", _this.directory.join(binPath)));

                          case 5:
                            _context2.next = 7;
                            return bin.chmod(binName, '755');

                          case 7:
                          case 'end':
                            return _context2.stop();
                        }
                      }
                    }, _callee2, _this);
                  }));

                  return function (_x) {
                    return _ref3.apply(this, arguments);
                  };
                }()));

              case 11:
              case 'end':
                return _context3.stop();
            }
          }
        }, _callee3, this);
      }));

      function linkBins() {
        return _ref2.apply(this, arguments);
      }

      return linkBins;
    }()
  }, {
    key: 'compile',
    value: function () {
      var _ref6 = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee5() {
        var _this2 = this;

        var distDirectory, packageJson, distPackageJson, _distPackageJson$bin, bin, shebang;

        return _regenerator2.default.wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                _context5.next = 2;
                return this.directory.rm('dist');

              case 2:
                _context5.next = 4;
                return this.directory.mkdir('dist');

              case 4:
                distDirectory = _context5.sent;
                _context5.next = 7;
                return this.packageJson();

              case 7:
                packageJson = _context5.sent;
                distPackageJson = new _packageJson2.default(packageJson).moveTo('src').toProduction().addBabelRuntime().toJSON();
                _context5.next = 11;
                return distDirectory.writeFile('package.json', distPackageJson);

              case 11:
                _context5.next = 13;
                return _promise2.default.all((packageJson.files || []).filter(function (fileName) {
                  return fileName.indexOf('src') !== 0;
                }).map(function (fileName) {
                  return distDirectory.cp(_this2.directory.join(fileName), fileName);
                }));

              case 13:
                _context5.next = 15;
                return this.directory.mkdir('src');

              case 15:
                _context5.next = 17;
                return this.directory.execSh(['\'' + require.resolve('babel-cli/bin/babel') + '\'', '--presets latest,stage-0', '--plugins transform-decorators-legacy,transform-runtime', '--copy-files', 'src', '--out-dir dist'].join(' '));

              case 17:
                _distPackageJson$bin = distPackageJson.bin;
                bin = _distPackageJson$bin === undefined ? {} : _distPackageJson$bin;
                shebang = '#!/usr/bin/env node';
                _context5.next = 22;
                return _promise2.default.all((0, _keys2.default)(bin).map(function () {
                  var _ref7 = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee4(binName) {
                    var binPath, binContents;
                    return _regenerator2.default.wrap(function _callee4$(_context4) {
                      while (1) {
                        switch (_context4.prev = _context4.next) {
                          case 0:
                            binPath = bin[binName];
                            _context4.next = 3;
                            return distDirectory.readFile(binPath);

                          case 3:
                            binContents = _context4.sent;
                            _context4.next = 6;
                            return distDirectory.writeFile(binPath, binContents.indexOf(shebang) === 0 ? binContents : shebang + '\n' + binContents);

                          case 6:
                            _context4.next = 8;
                            return distDirectory.chmod(binPath, '755');

                          case 8:
                          case 'end':
                            return _context4.stop();
                        }
                      }
                    }, _callee4, _this2);
                  }));

                  return function (_x2) {
                    return _ref7.apply(this, arguments);
                  };
                }()));

              case 22:
              case 'end':
                return _context5.stop();
            }
          }
        }, _callee5, this);
      }));

      function compile() {
        return _ref6.apply(this, arguments);
      }

      return compile;
    }()
  }]);
  return Project;
}();

exports.default = Project;
