// node-promise-es6/child-process.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.execFile = exports.exec = exports.execSync = exports.execFileSync = exports.spawnSync = exports.spawn = exports.fork = undefined;

var _child_process = require('child_process');

var _child_process2 = _interopRequireDefault(_child_process);

var _promisify = require('node-promise-es6/promisify');

var _promisify2 = _interopRequireDefault(_promisify);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// export * from 'child_process';
/* eslint-disable */
var fork = exports.fork = _child_process2.default.fork;
var spawn = exports.spawn = _child_process2.default.spawn;
var spawnSync = exports.spawnSync = _child_process2.default.spawnSync;
var execFileSync = exports.execFileSync = _child_process2.default.execFileSync;
var execSync = exports.execSync = _child_process2.default.execSync;
/* eslint-enable */

var exec = exports.exec = (0, _promisify2.default)(_child_process2.default.exec, ['stdout', 'stderr']);
var execFile = exports.execFile = (0, _promisify2.default)(_child_process2.default.execFile, ['stdout', 'stderr']);
