// node-promise-es6/fs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.read = exports.write = exports.access = exports.appendFile = exports.writeFile = exports.readFile = exports.fsync = exports.futimes = exports.utimes = exports.open = exports.close = exports.readdir = exports.mkdir = exports.rmdir = exports.unlink = exports.realpath = exports.readlink = exports.symlink = exports.link = exports.fstat = exports.lstat = exports.stat = exports.lchmod = exports.fchmod = exports.chmod = exports.lchown = exports.fchown = exports.chown = exports.truncate = exports.ftruncate = exports.rename = exports.FileWriteStream = exports.WriteStream = exports.createWriteStream = exports.FileReadStream = exports.ReadStream = exports.createReadStream = exports.realpathSync = exports.unwatchFile = exports.watchFile = exports.watch = exports.appendFileSync = exports.writeFileSync = exports.futimesSync = exports.utimesSync = exports.chownSync = exports.fchownSync = exports.chmodSync = exports.fchmodSync = exports.unlinkSync = exports.linkSync = exports.symlinkSync = exports.readlinkSync = exports.statSync = exports.lstatSync = exports.fstatSync = exports.readdirSync = exports.mkdirSync = exports.fsyncSync = exports.rmdirSync = exports.ftruncateSync = exports.truncateSync = exports.renameSync = exports.writeSync = exports.readSync = exports.openSync = exports.closeSync = exports.readFileSync = exports.accessSync = exports.X_OK = exports.W_OK = exports.R_OK = exports.F_OK = exports.Stats = undefined;

var _fs = require('fs');

var _fs2 = _interopRequireDefault(_fs);

var _promisify = require('node-promise-es6/promisify');

var _promisify2 = _interopRequireDefault(_promisify);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// export * from 'fs';
/* eslint-disable */
var Stats = exports.Stats = _fs2.default.Stats;
var F_OK = exports.F_OK = _fs2.default.F_OK;
var R_OK = exports.R_OK = _fs2.default.R_OK;
var W_OK = exports.W_OK = _fs2.default.W_OK;
var X_OK = exports.X_OK = _fs2.default.X_OK;
var accessSync = exports.accessSync = _fs2.default.accessSync;
var readFileSync = exports.readFileSync = _fs2.default.readFileSync;
var closeSync = exports.closeSync = _fs2.default.closeSync;
var openSync = exports.openSync = _fs2.default.openSync;
var readSync = exports.readSync = _fs2.default.readSync;
var writeSync = exports.writeSync = _fs2.default.writeSync;
var renameSync = exports.renameSync = _fs2.default.renameSync;
var truncateSync = exports.truncateSync = _fs2.default.truncateSync;
var ftruncateSync = exports.ftruncateSync = _fs2.default.ftruncateSync;
var rmdirSync = exports.rmdirSync = _fs2.default.rmdirSync;
var fsyncSync = exports.fsyncSync = _fs2.default.fsyncSync;
var mkdirSync = exports.mkdirSync = _fs2.default.mkdirSync;
var readdirSync = exports.readdirSync = _fs2.default.readdirSync;
var fstatSync = exports.fstatSync = _fs2.default.fstatSync;
var lstatSync = exports.lstatSync = _fs2.default.lstatSync;
var statSync = exports.statSync = _fs2.default.statSync;
var readlinkSync = exports.readlinkSync = _fs2.default.readlinkSync;
var symlinkSync = exports.symlinkSync = _fs2.default.symlinkSync;
var linkSync = exports.linkSync = _fs2.default.linkSync;
var unlinkSync = exports.unlinkSync = _fs2.default.unlinkSync;
var fchmodSync = exports.fchmodSync = _fs2.default.fchmodSync;
var chmodSync = exports.chmodSync = _fs2.default.chmodSync;
var fchownSync = exports.fchownSync = _fs2.default.fchownSync;
var chownSync = exports.chownSync = _fs2.default.chownSync;
var utimesSync = exports.utimesSync = _fs2.default.utimesSync;
var futimesSync = exports.futimesSync = _fs2.default.futimesSync;
var writeFileSync = exports.writeFileSync = _fs2.default.writeFileSync;
var appendFileSync = exports.appendFileSync = _fs2.default.appendFileSync;
var watch = exports.watch = _fs2.default.watch;
var watchFile = exports.watchFile = _fs2.default.watchFile;
var unwatchFile = exports.unwatchFile = _fs2.default.unwatchFile;
var realpathSync = exports.realpathSync = _fs2.default.realpathSync;
var createReadStream = exports.createReadStream = _fs2.default.createReadStream;
var ReadStream = exports.ReadStream = _fs2.default.ReadStream;
var FileReadStream = exports.FileReadStream = _fs2.default.FileReadStream;
var createWriteStream = exports.createWriteStream = _fs2.default.createWriteStream;
var WriteStream = exports.WriteStream = _fs2.default.WriteStream;
var FileWriteStream = exports.FileWriteStream = _fs2.default.FileWriteStream;
/* eslint-enable */

var rename = exports.rename = (0, _promisify2.default)(_fs2.default.rename);
var ftruncate = exports.ftruncate = (0, _promisify2.default)(_fs2.default.ftruncate);
var truncate = exports.truncate = (0, _promisify2.default)(_fs2.default.truncate);
var chown = exports.chown = (0, _promisify2.default)(_fs2.default.chown);
var fchown = exports.fchown = (0, _promisify2.default)(_fs2.default.fchown);
var lchown = exports.lchown = (0, _promisify2.default)(_fs2.default.lchown);
var chmod = exports.chmod = (0, _promisify2.default)(_fs2.default.chmod);
var fchmod = exports.fchmod = (0, _promisify2.default)(_fs2.default.fchmod);
var lchmod = exports.lchmod = (0, _promisify2.default)(_fs2.default.lchmod);
var stat = exports.stat = (0, _promisify2.default)(_fs2.default.stat);
var lstat = exports.lstat = (0, _promisify2.default)(_fs2.default.lstat);
var fstat = exports.fstat = (0, _promisify2.default)(_fs2.default.fstat);
var link = exports.link = (0, _promisify2.default)(_fs2.default.link);
var symlink = exports.symlink = (0, _promisify2.default)(_fs2.default.symlink);
var readlink = exports.readlink = (0, _promisify2.default)(_fs2.default.readlink);
var realpath = exports.realpath = (0, _promisify2.default)(_fs2.default.realpath);
var unlink = exports.unlink = (0, _promisify2.default)(_fs2.default.unlink);
var rmdir = exports.rmdir = (0, _promisify2.default)(_fs2.default.rmdir);
var mkdir = exports.mkdir = (0, _promisify2.default)(_fs2.default.mkdir);
var readdir = exports.readdir = (0, _promisify2.default)(_fs2.default.readdir);
var close = exports.close = (0, _promisify2.default)(_fs2.default.close);
var open = exports.open = (0, _promisify2.default)(_fs2.default.open);
var utimes = exports.utimes = (0, _promisify2.default)(_fs2.default.utimes);
var futimes = exports.futimes = (0, _promisify2.default)(_fs2.default.futimes);
var fsync = exports.fsync = (0, _promisify2.default)(_fs2.default.fsync);
var readFile = exports.readFile = (0, _promisify2.default)(_fs2.default.readFile);
var writeFile = exports.writeFile = (0, _promisify2.default)(_fs2.default.writeFile);
var appendFile = exports.appendFile = (0, _promisify2.default)(_fs2.default.appendFile);
var access = exports.access = (0, _promisify2.default)(_fs2.default.access);
var write = exports.write = (0, _promisify2.default)(_fs2.default.write, ['written', 'data']);
var read = exports.read = (0, _promisify2.default)(_fs2.default.read, ['bytesRead', 'buffer']);
